src/cargo/core/features.rs

```diff
@@ -488,7 +497,9 @@ impl Features {
             let feature = feature.name.replace("_", "-");
             let mut msg = format!("feature `{}` is required", feature);
 
-            if nightly_features_allowed() {
+            // TODO
```

Comment:
`Features::new` takes a `Config`; could you just read and cache `nightly_features_allowed` in the constructor, so that it doesn't have to be passed to every `require`?

Reply:
Ooh, good idea, thanks! I had it stuck in my head that I needed a whole `Config` for some reason.

Reply:
Now that you've changed that, do you still need to pass `config` to `Features::add`, or could you use the same cached value?

Reply:
I can use the cached value, I updated it.